examples/rich-text-presence/client.js

```diff
@@ -0,0 +1,142 @@
+var ReconnectingWebSocket = require('reconnecting-websocket');
+var sharedb = require('sharedb/lib/client');
+var richText = require('./rich-text');
+var Quill = require('quill');
+var QuillCursors = require('quill-cursors');
+var tinycolor = require('tinycolor2');
+
+sharedb.types.register(richText.type);
+Quill.register('modules/cursors', QuillCursors);
+
+var clients = [];
+var colors = {};
+
+var collection = 'examples';
+var id = 'richtext';
+
+var editorContainer = document.querySelector('.editor-container');
+document.querySelector('#add-client').addEventListener('click', function() {
+  addClient();
+});
+
+addClient();
+addClient();
+
+function addClient() {
+  var socket = new ReconnectingWebSocket('ws://' + window.location.host);
+  var connection = new sharedb.Connection(socket);
+  var doc = connection.get(collection, id);
+  doc.subscribe(function(err) {
+    if (err) throw err;
+    var quill = initialiseQuill(doc);
+    var color = '#' + tinycolor.random().toHex();
+    var id = 'client-' + (clients.length + 1);
```

Comment:
As a note, the `presenceId`s being fixed in this demo means that opening the demo in two different tabs can result in some pretty weird stuff happening.

I didn't dig into it too much, but I suspect that "client-1" in tab A and "client-1" in tab B were both trying to update the same presence, and the displayed presence in the other clients was changing based on which tab had a higher `seq`. I think we did discuss that last week, with a potential solution being to prepend a generated client id like "original src".

Reply:
Ha yes I hadn't tried multiple tabs. To be fair, maybe just showing one client per tab and just adding some instructional text to open a new tab to test is easiest. It would:

- fix this issue
- simplify example code (always a good thing)
- be a bit more "realistic"

What do you think?

Reply:
I've gone ahead and simplified it.